src/fixers/misc_tracker.ts

```diff
@@ -0,0 +1,348 @@
+import { MSG_TYPE_SAVE_SETTINGS } from "../constants";
+import { sendToServiceWorker } from "../comms";
+
+export class MiscTracker implements IMutationAware, IStateAware {
+
+    private displayMiscTracker = true;
+    private miscQualities: Map<string, TrackedQuality> = new Map();
+    private qualityNameAndCategory: Map<string, string> = new Map();
+    private qualityNames: string[] = [];
+    private currentState?: GameState;
+    private settingsString: string | undefined;
+    private currentSettings: SettingsObject | undefined;
+
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    constructor() {
+
+    }
+
+    applySettings(settings: SettingsObject): void {
+        this.currentSettings = settings;
+        console.log("received settings")
+        console.log(settings)
+        let temp = settings.trackedQualities as string;
+        if (!temp) {
+            temp = "{}";
+        }
+        this.miscQualities = new Map(Object.entries(JSON.parse(temp)));
+    }
+
+    
+
+    linkState(state: GameStateController): void {
+        state.onCharacterDataLoaded((g) => {
+            this.currentState = g;
+            const unsortedQualityNames: string[] = []
+            for (const quality of g.enumerateQualities()) {
+                this.qualityNameAndCategory.set(quality.name, quality.category);
+                unsortedQualityNames.push(quality.name);
+            }
+            this.qualityNames = unsortedQualityNames.sort(this.stringSorter())
+            for (const [key, value] of this.miscQualities) {
+                const quality = g.getQuality(value.category, key);
+                if (quality) {
+                    value.currentValue = quality.level;
+                } else {
+                    value.currentValue = 0;
+                }
+
+                this.updateTracker(key, quality?.level || 0);
+            }
+        });
+
+        state.onQualityChanged((state, quality, _previous, current) => {
+            if (!this.qualityNames.includes(quality.name)) {
+                this.qualityNames.push(quality.name);
+                this.qualityNames = this.qualityNames.sort(this.stringSorter())
+                this.qualityNameAndCategory.set(quality.name, quality.category);
+
+                const miscSelect = document.getElementById("track-target-name");
+                const option = document.createElement("option");
+                option.value = quality.name;
+                option.text = quality.name;
+                miscSelect?.appendChild(option);
+                
+            }
+            if (this.miscQualities.has(quality.name)) {
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                this.miscQualities.get(quality.name)!.currentValue = current;
+                this.updateTracker(quality.name, current);
+            }
+        });
+    }
+
+    private stringSorter(): ((a: string, b: string) => number) | undefined {
```

Comment:
I believe that could be replaced with `const stringSorter = (s1, s2) => (s1 > s2 ? -1 : 1)`